File: src/main/java/it/uniud/newbestsub/utils/BestSubsetFormatter.java
package it.uniud.newbestsub.utils;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.logging.*;

class BestSubsetFormatter extends Formatter {

    private static final DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy hh:mm:ss.SSS");

    public String format(LogRecord record) {
        StringBuilder builder = new StringBuilder(1000);
        builder.append(formatMessage(record));
        builder.append(" - TIME: ").append(dateFormat.format(new Date(record.getMillis())));
        builder.append("\n");
        return builder.toString();
    }

    public String getHead(Handler h) {
        return super.getHead(h);
    }

    public String getTail(Handler h) {
        return super.getTail(h);
    }
}

File: src/main/kotlin/it/uniud/newbestsub/utils/BestSubsetFormatter.kt
package it.uniud.newbestsub.utils

import java.text.SimpleDateFormat
import java.util.Date
import java.util.logging.*

class BestSubsetFormatter : Formatter() {

    override fun format(record: LogRecord): String {
        val dateFormat = SimpleDateFormat("dd/MM/yyyy hh:mm:ss.SSS")
        val builder = StringBuilder(1000)
        builder.append(formatMessage(record))
        builder.append(" - TIME: ").append(dateFormat.format(Date(record.millis)))
        builder.append("\n")
        return builder.toString()
    }

    override fun getHead(h: Handler?): String {
        return super.getHead(h)
    }

    override fun getTail(h: Handler?): String {
        return super.getTail(h)
    }

}
